core/src/types/internal.rs

```diff
@@ -48,12 +48,18 @@ impl From<bool> for HostEnvResult {
 mod tx_queue {
     use borsh::{BorshDeserialize, BorshSerialize};
 
+    use crate::types::transaction::encrypted::EncryptedTx;
+
     /// A wrapper for `crate::types::transaction::WrapperTx` to conditionally
     /// add `has_valid_pow` flag for only used in testnets.
     #[derive(Debug, Clone, BorshDeserialize, BorshSerialize)]
     pub struct WrapperTxInQueue {
         /// Wrapper tx
         pub tx: crate::types::transaction::WrapperTx,
+        /// the encrypted payload
+        pub inner_tx: Option<EncryptedTx>,
+        /// the encrypted payload
+        pub inner_tx_code: Option<EncryptedTx>,
```

Comment:
Given that this type can no also be code, maybe it should be renamed to something more generic. Maybe even just `Encrypted`.

Reply:
Just renamed `WrapperTxInQueue` to `TxInQueue` to represent that all 3 fields are included in  and are the main constituents of the `Tx` struct.